CardScan/Classes/CreditCardUtils.swift

```diff
@@ -25,15 +41,12 @@ public struct CreditCardUtils {
     }
     
     public static func isValidBin(number: String) -> Bool {
-        return isAmex(number: number) || isDiscover(number: number) || isVisa(number: number) || isMastercard(number: number) || isUnionPay(number: number)
+        return isAmex(number: number) || isDiscover(number: number) || isVisa(number: number) || isMastercard(number: number) || isUnionPay(number: number) || isJcb(number: number) || isDinersClub(number: number)
     }
     
     public static func isAmex(number: String) -> Bool {
```

Comment:
Can we get rid of these methods? Instead of specific `isNetwork` methods, we should simplify the methods using these.

For example, if we are using these methods to set an icon
```
if (CreditCardUtils.isAmex()) {
   setIcon(amexIcon)
} else if (CreditCardUtils.isUnionPay()) {
  setIcon(unionPay)
} else if ...
```

we should instead create a method in CreditCardUtils that returns the desired icon.
```
switch (determineCardNetwork(number)) {
    case VISA -> visaIcon
    case AMEX -> amexIcon
    ...
}
```

Reply:
Centralizing into one `determineCardNetwork` method also keeps all the prefix logic in one place instead of spread out around multiple methods

see https://github.com/getbouncer/cardscan-android/blob/master/cardscan-base/src/main/java/com/getbouncer/cardscan/base/CreditCardUtils.kt#L159

Reply:
this _also_ lets us more easily deal with prioritizing discover over unionpay when BINs overlap.